packages/react/src/time-input/utils.test.ts

```diff
@@ -0,0 +1,166 @@
+import { isValidTime, formatTime } from './utils';
+
+describe('isValidTime', () => {
+	test('Valid values', () => {
+		expect(isValidTime('0')).toEqual(true);
+		expect(isValidTime('1')).toEqual(true);
+		expect(isValidTime('11')).toEqual(true);
+		expect(isValidTime('111')).toEqual(true);
+		expect(isValidTime('1111')).toEqual(true);
+		expect(isValidTime('11111')).toEqual(true);
+		expect(isValidTime('111111')).toEqual(true);
+		expect(isValidTime('21')).toEqual(true);
+		expect(isValidTime('2100')).toEqual(true);
+		expect(isValidTime('21:00')).toEqual(true);
+		expect(isValidTime('21:00:00')).toEqual(true);
+		expect(isValidTime('24')).toEqual(true);
+		expect(isValidTime('9')).toEqual(true);
+		expect(isValidTime('9:00')).toEqual(true);
+		expect(isValidTime('9am')).toEqual(true);
+		expect(isValidTime('9pm')).toEqual(true);
+		expect(isValidTime('9 am')).toEqual(true);
+		expect(isValidTime('9 pm')).toEqual(true);
+		expect(isValidTime("9 o'clock")).toEqual(true);
+	});
+
+	test('Invalid values', () => {
+		expect(isValidTime('')).toEqual(false);
+		expect(isValidTime('666')).toEqual(false);
+		expect(isValidTime('6666')).toEqual(false);
+		expect(isValidTime('66666')).toEqual(false);
+		expect(isValidTime('666666')).toEqual(false);
+		expect(isValidTime('25')).toEqual(false);
+		expect(isValidTime('2500')).toEqual(false);
+		expect(isValidTime('25:00')).toEqual(false);
+		expect(isValidTime('25:00:00')).toEqual(false);
+		expect(isValidTime('9:66')).toEqual(false);
+		expect(isValidTime('pm')).toEqual(false);
+		expect(isValidTime('am')).toEqual(false);
+	});
+});
+
+describe('formatTime', () => {
```

Comment:
Some other test ideas to check format doesn't break:

- T03:21:48.433Z
- 14:24:30
- 2:22:33 pm
- robust 24 hr to 12 hr time tests (a loop similar to what you did on line 60)

Reply:
> * `T03:21:48.433Z`

Done: https://github.com/agriculturegovau/agds-next/pull/1582/commits/9f041393d7d5431ec7b0b054d4fa17df8a46bef5#diff-a028502510a88730d9fb1b694d8881dafc37d045dfa2e0d0e881c59e669ebdb2R167

> * `2024-04-04T03:32:49.648Z` 

Won't support

> * `14:24:30`

Done: https://github.com/agriculturegovau/agds-next/pull/1582/commits/9f041393d7d5431ec7b0b054d4fa17df8a46bef5#diff-a028502510a88730d9fb1b694d8881dafc37d045dfa2e0d0e881c59e669ebdb2R15

> * `2:22:33 pm`

Done: https://github.com/agriculturegovau/agds-next/pull/1582/commits/9f041393d7d5431ec7b0b054d4fa17df8a46bef5#diff-a028502510a88730d9fb1b694d8881dafc37d045dfa2e0d0e881c59e669ebdb2R160

> * robust 24 hr to 12 hr time tests (a loop similar to what you did on line 60)

I feel like it's already robust enough

> * `1540` 

Already existed: https://github.com/agriculturegovau/agds-next/pull/1582/commits/9f041393d7d5431ec7b0b054d4fa17df8a46bef5#diff-a028502510a88730d9fb1b694d8881dafc37d045dfa2e0d0e881c59e669ebdb2R107